tangos/input_handlers/ramsesHOP.py

```diff
@@ -0,0 +1,137 @@
+        return np.array([adaptahop_halo.properties['x'], adaptahop_halo.properties['y'], adaptahop_halo.properties['z']])
+
+    @staticmethod
+    def _compute_contamination_fraction(adaptahop_halo):
+        return float(adaptahop_halo.properties['ntot_contam'] / adaptahop_halo.properties['ntot'])
+
+    def available_object_property_names_for_timestep(self, ts_extension, object_typetag):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        halo_attributes = list(h._halo_attributes)
+        if h._read_contamination:
+            halo_attributes.extend(h._halo_attributes_contam)
+
+        attrs = chain.from_iterable(tuple(always_iterable(attr)) for (attr, _len, _dtype) in halo_attributes)
+
+        # Import all precalculated properties except conflicting ones with Tangos syntax
+        property_list = [attr for attr in attrs if attr not in self._exclude_adaptahop_precalculated_properties()]
+        # Add additional properties that are baisc to most Tangos databases derived
+        property_list += self._include_additional_properties_derived_from_adaptahop()
+        return property_list
+
+    @staticmethod
+    def _resolve_units(value):
+        import pynbody
+        # TODO Solve the fact that AdapataHOP stores distances in Mpc, when Tangos physical units are kpc
+        if (pynbody.units.is_unit(value)):
+            return float(value)
+        else:
+            return value
+
+    def _get_map_child_subhalos(self, ts_extension):
+        h = self._construct_halo_cat(ts_extension, 'halo')
+        halo_children = {}
+        for halo_i in range(1, len(h)+1):  # AdaptaHOP catalogues start at 1
+            halo_props = h[halo_i].properties
+           
+            if halo_props['host_id'] != halo_i: # If halo isn't its own host, it is a subhalo
+                parent = halo_props['host_id']
+                if parent not in halo_children:
+                    halo_children[parent] = []
+                halo_children[parent].append(halo_i)
+        return halo_children
+
+    def iterate_object_properties_for_timestep(self, ts_extension, object_typetag, property_names):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        if "child" in property_names:
+            # Construct the mapping between parent and subhalos
+            map_child_parent = self._get_map_child_subhalos(ts_extension)
+            print(map_child_parent)
+    
+        for halo_i in range(1, len(h)+1):  # AdaptaHOP catalogues start at 1
+
+            # Tangos expects data to have a finder offset, and a finder id following the stat file logic
+            # I think these are irrelevant for AdaptaHOP catalogues which are derived directly from pynbody
+            # Putting the finder ID twice seems to produce consistent results
+            all_data = [halo_i, halo_i]
+
+            # Loop over all properties we wish to import
+            for k in property_names:
+
+                adaptahop_halo = h[halo_i]
```

Comment:
Shouldn't this be moved outside the loop?

Reply:
Indeed, this should now be fixed in 0d60675